pkg/apis/config/feature_flags_test.go

```diff
@@ -305,6 +305,28 @@ func TestCheckAlphaOrBetaAPIFields(t *testing.T) {
 	}
 }
 
+func TestIsSpireEnabled(t *testing.T) {
+	ctx := context.Background()
+	if config.IsSpireEnabled(ctx) {
+		t.Errorf("IsSpireEnabled got true but expected to be false")
+	}
+	store := config.NewStore(logging.FromContext(ctx).Named("config-store"))
+	featureflags := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "feature-flags",
+		},
+		Data: map[string]string{
+			"enable-api-fields":         "alpha",
```

Comment:
noticed that we have `enable-api-fields` set to `"alpha"` here, but then the function `IsSpireEnabled` doesn't check for it 🤔

Reply:
If enable-api-fields is not set to alpha then FeatureFlags.EnforceNonFalsifiability will be set to the default value, which is no enforcement. This is done at the time reading the featureflags configmap.
https://github.com/tektoncd/pipeline/blob/abd184916d4caef9e60c942eed194afd02874106/pkg/apis/config/feature_flags.go#L200

As such we do not need to check the enable-api-fields here.

Reply:
> will be set to the default value, which is no enforcement

does this mean we will change the user-specified config? I'd expect that we'd throw a validation error?

Reply:
i found that we override the value to None https://github.com/tektoncd/pipeline/blob/abd184916d4caef9e60c942eed194afd02874106/pkg/apis/config/feature_flags.go#L213

then we confirm that it matches that the default value which is None 🤔 

https://github.com/tektoncd/pipeline/blob/abd184916d4caef9e60c942eed194afd02874106/pkg/apis/config/feature_flags.go#L214-L216

Reply:
please add several test cases here:
- "enforce-nonfalsifiability" is set to "spire" and "enable-api-fields" set to "alpha" --> expecting true
- "enforce-nonfalsifiability" is set to "spire" and "enable-api-fields" set to "stable"/"beta" --> seems current implementation will change "enforce-nonfalsifiability" to "none" so expecting false 🤔 
- "enforce-nonfalsifiability" is set to "none" and "enable-api-fields" set to "alpha" --> expecting false
- "enforce-nonfalsifiability" is set to "none" and "enable-api-fields" set to "stable"/"beta" --> expecting false

Reply:
also please update the docs to say that alpha feature flag must also be enabled (can be added to item 1) - https://github.com/tektoncd/pipeline/blob/main/docs/spire.md#enabling-taskrun-result-attestations

Reply:
Error is returned here
https://github.com/tektoncd/pipeline/blob/abd184916d4caef9e60c942eed194afd02874106/pkg/apis/config/feature_flags.go#L215

Reply:
if we return an error, why do we change the value in line 213?
https://github.com/tektoncd/pipeline/blob/abd184916d4caef9e60c942eed194afd02874106/pkg/apis/config/feature_flags.go#L213

Reply:
either way, please add those test cases to `TestIsSpireEnabled` 🙏🏾

Reply:
@jagathprakash opened https://github.com/tektoncd/pipeline/pull/6527 to address my comments above, ptal

Reply:
Added tests for all conditions.